src/components/Header/Header.js

```diff
@@ -0,0 +1,35 @@
+import React, {Component} from 'react';
+import { Link } from "react-router-dom";
+
+import Button from '../Common/Button';
+import './header.css';
+
+export default class Header extends Component {
+    constructor(props) {
+        super(props);
+        this.searchClick = this.searchClick.bind(this);
+    }
+    searchClick(ev) {
+        ev.preventDefault();
+    }
+
+    render() {
+        var view = '';
+        if (this.props.showSearchButton === true) {
+            view = (

```

Comment:
This should be another component without any problem, will be easier to write tests for this.
In this case, you can use the `children` prop to render optional stuffs instead of creating flags for it.

```javascript
render() {
   const { children } = this.props;

   return (
      <div> <nav> <a> </a> {children} </nav> </div>
   );
}
```

This way, you can re-use this header event when you don't have this search bar, maybe create another component to hold the implementation if you use it more than once.

Reply:
Extracted the view into another component.